<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>944</width>
    <height>548</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Course2 Practice - Pathfinder</string>
  </property>
  <property name="windowIcon">
   <iconset resource="assets.qrc">
    <normaloff>:/assets/path.png</normaloff>:/assets/path.png</iconset>
  </property>
  <widget class="QWidget" name="layoutCentral">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <widget class="QWidget" name="layoutButtons" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <property name="spacing">
        <number>5</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QPushButton" name="btnAdd">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Начать создание препятствия.&lt;/span&gt; Нажатием мыши последовательно создаются вершины многоугольника, формирующие собой препятствие&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;ЛКМ:&lt;/span&gt; Добавить точку по координатам мыши&lt;br/&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;ПКМ:&lt;/span&gt; Подтвердить создание препятствия и перейти к установке непроходимости&lt;br/&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Shift+ЛКМ:&lt;/span&gt; Убрать последнюю добавленную точку&lt;br/&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Shift+ПКМ:&lt;/span&gt; Выйти из режима создания препятствия&lt;/p&gt;&lt;p&gt;Горячая клавиша:&lt;span style=&quot; font-weight:700;&quot;&gt; [ 2 ]&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="styleSheet">
          <string notr="true">text-align: left;
padding: 3px;</string>
         </property>
         <property name="text">
          <string>Добавить</string>
         </property>
         <property name="icon">
          <iconset resource="assets.qrc">
           <normaloff>:/assets/add.png</normaloff>:/assets/add.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnDelete">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Начать удаление препятствия.&lt;/span&gt; Нажатием мыши удалить препятствие, которое находится под курсором мыши&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;ЛКМ:&lt;/span&gt; Удалить препятствие под курсором&lt;br/&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;ПКМ:&lt;/span&gt; Выйти из режима удаления препятствия&lt;/p&gt;&lt;p&gt;Горячая клавиша:&lt;span style=&quot; font-weight:700;&quot;&gt; [ 3 ]&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="styleSheet">
          <string notr="true">text-align: left;
padding: 3px;</string>
         </property>
         <property name="text">
          <string>Удалить</string>
         </property>
         <property name="icon">
          <iconset resource="assets.qrc">
           <normaloff>:/assets/erase.png</normaloff>:/assets/erase.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnEdit">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Начать редактирование препятствия.&lt;/span&gt; При редактировании мышью можно редактировать препятствия на карте&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;ЛКМ &lt;/span&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;(зажать)&lt;/span&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;:&lt;/span&gt; Нажмите на точку и перетаскивайте для её перемещения по карте&lt;br/&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;ПКМ:&lt;/span&gt; Выйти из режима редактирования препятствия&lt;br/&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Shift+ЛКМ:&lt;/span&gt; Добавить точку в препятствие под курсором мыши. Точка будет добавлена к ближайшему лежащему ребру. Курсор мыши должен быть внутри препятствия&lt;br/&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Shift+ПКМ:&lt;/span&gt; Удалить точку, находящуюся под курсором мыши&lt;/p&gt;&lt;p&gt;Горячая клавиша:&lt;span style=&quot; font-weight:700;&quot;&gt; [ 4 ]&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="styleSheet">
          <string notr="true">text-align: left;
padding: 3px;</string>
         </property>
         <property name="text">
          <string>Редактировать</string>
         </property>
         <property name="icon">
          <iconset resource="assets.qrc">
           <normaloff>:/assets/edit.png</normaloff>:/assets/edit.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnStart">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Задать начальную точку маршрута.&lt;/span&gt; Нажатием мыши задаётся начальная точка маршрута&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;ЛКМ:&lt;/span&gt; Установить точку под курсором мыши &lt;br/&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;ПКМ:&lt;/span&gt; Выйти из режима задания точки&lt;/p&gt;&lt;p&gt;Горячая клавиша:&lt;span style=&quot; font-weight:700;&quot;&gt; [ 5 ]&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="styleSheet">
          <string notr="true">text-align: left;
padding: 3px;</string>
         </property>
         <property name="text">
          <string>Старт</string>
         </property>
         <property name="icon">
          <iconset resource="assets.qrc">
           <normaloff>:/assets/start.png</normaloff>:/assets/start.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnFinish">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Задать конечную точку маршрута.&lt;/span&gt; Нажатием мыши задаётся конечная точка маршрута&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;ЛКМ:&lt;/span&gt; Установить точку под курсором мыши &lt;br/&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;ПКМ:&lt;/span&gt; Выйти из режима задания точки&lt;/p&gt;&lt;p&gt;Горячая клавиша:&lt;span style=&quot; font-weight:700;&quot;&gt; [ 6 ]&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="styleSheet">
          <string notr="true">text-align: left;
padding: 3px;</string>
         </property>
         <property name="text">
          <string>Финиш</string>
         </property>
         <property name="icon">
          <iconset resource="assets.qrc">
           <normaloff>:/assets/finish.png</normaloff>:/assets/finish.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="spacerButtons">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="btnLoad">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Загрузить карту из XML-файла.&lt;/span&gt; Загрузить карту из XML-файла, который лежит вместе с программой&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="styleSheet">
          <string notr="true">text-align: left;
padding: 3px;</string>
         </property>
         <property name="text">
          <string>Загрузить карту</string>
         </property>
         <property name="icon">
          <iconset resource="assets.qrc">
           <normaloff>:/assets/load.png</normaloff>:/assets/load.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnSave">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Сохранить карту в XML-файл.&lt;/span&gt; Сохранить карту в XML-файл рядом с программой&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="styleSheet">
          <string notr="true">text-align: left;
padding: 3px;</string>
         </property>
         <property name="text">
          <string>Сохранить карту</string>
         </property>
         <property name="icon">
          <iconset resource="assets.qrc">
           <normaloff>:/assets/save.png</normaloff>:/assets/save.png</iconset>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="layoutMain" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="Canvas" name="widgetGraph" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>800</width>
           <height>500</height>
          </size>
         </property>
        </widget>
       </item>
       <item alignment="Qt::AlignBottom">
        <widget class="QWidget" name="layoutStatus" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="labelStatus">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Строка статуса.&lt;/span&gt; Здесь будут появляться результаты действий, таких как добавление, удаление и т.д.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Нет работы</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="spacerStatus">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="labelSize">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Изменить размеры карты.&lt;/span&gt; Задать размеры карты.&lt;/p&gt;&lt;p&gt;Примечание: все объекты, а также путь, будут удалены&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>800 x 500</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelObjects">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Счётчик объектов.&lt;/span&gt; Показывает сколько существует препятствий на карте&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>0 объектов</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line1">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelCoords">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Координаты мыши на карте.&lt;/span&gt; Показывает координаты курсора мыши на карте.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>X, Y: [-, -]</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Canvas</class>
   <extends>QWidget</extends>
   <header>canvas.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="assets.qrc"/>
 </resources>
 <connections/>
</ui>
